packages/orm/src/index.ts

```diff
@@ -437,3 +438,34 @@ export function getConstructorFromTaggedId(id: string): EntityConstructor<any> {
 function equal(a: any, b: any): boolean {
   return a === b || (a instanceof Date && b instanceof Date && a.getTime() == b.getTime());
 }
+
+// The Entity type doesn't model the `changes` structure yet, so do this for now.
+interface EntityChanges<T> {
+  // Ideally we could use Record<keyof T, { hasChanged: boolean }> but we only want
+  // keys that are actually columns. Right now OptsOf<T> has collections (not in .changed)
+  // and OrderOf<T> has id/createdAt/updatedAt (also not in .changed), so just using an index
+  // type for now.
+  changes: { [key: string]: { hasChanged: boolean } };
+}
+
+const cannotBeChangedRule = Symbol("cannotBeChangedRule");
```

Comment:
praise: Neat. I haven't seen this use of `Symbol` like this before. I think I `get it` and how it's used, but would also appreciate a @stephenh thought process dump.

Reply:
@chr1sjf0x sure, I think it just ensures no conflicts on keys.

Like if we picked a string-based key like `"foo"` but then, in the invariably large/annoying complexities of ~1000-deps-in-`node_modules`, some _other_ library also used `"foo"` for its metadata, we'd stomp on each other.

The symbol just ensures a unique/namespaced lookup on the object.